micro-rdk/src/common/registry.rs

```diff
@@ -48,11 +48,11 @@ pub fn get_board_from_dependencies(deps: Vec<Dependency>) -> Option<BoardType> {
 // first element is a string representing the component type (arm, motor, etc.)
 // and the second element is its name.
 #[derive(Debug, Clone, Hash, Eq, PartialEq)]
-pub struct ResourceKey(pub &'static str, pub String);
+pub struct ResourceKey(pub String, pub String);
```

Comment:
If the below `ResourceKey::new` is truly unused, could we introduce a new `new` that admitted both argument types:

```
impl ResourceKey {
    pub fn new(model: impl Into<String>, name: impl Into<String>) -> Self {
        Self ( model.into(), name.into() )
    }
}
```

Then you could drop the `.to_string` calls on, e.g.:

```
let r_key = ResourceKey(EncoderCompName.to_string(), enc_name);
```